packages/js/data/src/crud/actions.ts

```diff
@@ -183,20 +183,25 @@ export const createDispatchActions = ( {
 
 	const updateItem = function* (
 		idQuery: IdQuery,
-		query: Partial< ItemQuery >
+		data: Partial< ItemQuery >
 	) {
-		const urlParameters = getUrlParameters( namespace, idQuery );
-		const { id, key } = parseId( idQuery, urlParameters );
+		const { key } = parseId(
+			idQuery,
+			getUrlParameters( namespace, idQuery )
+		);
 
 		try {
+			const templatePath = `${ namespace }/{id}`;
+			const urlParameters = getUrlParameters( templatePath, idQuery );
```

Comment:
Do we need to call `getUrlParameters` twice in this method?

Reply:
Yes, I tried to use it once but the key was wrongly generated, also note the params received by the `getUrlParameters` are different.